packages/sdk/src/utils/index.ts

```diff
@@ -0,0 +1,360 @@
+    if (isObject(value)) {
+      // eslint-disable-next-line no-param-reassign
+      obj[key] = encodeDecodeObject(value as NestedObject, {
+        encode,
+        depth: depth + 1,
+      });
+    } else if (isString(value)) {
+      // eslint-disable-next-line no-param-reassign
+      obj[key] = encode
+        ? encodeURIComponent(value as string)
+        : decodeURIComponent(value as string);
+    }
+  }
+
+  return obj;
+}
+
+// TODO: Make this function immutable
+/**
+ * Encodes an object into [valid URI components](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent),
+ * modifying the object in the process.
+ *
+ * @param obj Object
+ * @returns The object is mutated.
+ * @deprecated
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function UNSTABLE_encodeObject(obj: NestedObject) {
+  return encodeDecodeObject(obj, { encode: true });
+}
+
+// TODO: Make this function immutable
+/**
+ * Decodes an object into [valid URI components](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent),
+ * modifying the object in the process.
+ *
+ * @param obj Object
+ * @returns The object is mutated.
+ * @deprecated
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function UNSTABLE_decodeObject(obj: NestedObject) {
+  return encodeDecodeObject(obj, { encode: false });
+}
+
+// Temporary convertors until bignumber.js is integrated
+export function convertSatoshisToBTC(satoshis: number) {
+  return satoshis / 10 ** 8;
+}
+
+export function convertBTCToSatoshis(btc: number) {
+  return parseInt((btc * 10 ** 8).toString(), 10); // remove floating point overflow by parseInt
+}
+
+export function generateTxUniqueIdentifier(txId: string, index: number) {
+  return `${txId}:${index}`;
+}
+
+/**
+ * Converts an outpoint to id format.
+ *
+ * An outpoint refers to a specific output.
+ *
+ * Reference: https://developer.bitcoin.org/reference/transactions.html#outpoint-the-specific-part-of-a-specific-output
+ *
+ * @param outpoint Outpoint in string format `{txid}:{vout}`
+ * @returns id formatted string, `{txid}i{vout}`
+ */
+export function outpointToIdFormat(outpoint: string) {
+  if (outpoint.includes(":")) {
+    return outpoint.replace(":", "i");
+  }
+
+  return outpoint.includes("i") ? outpoint : `${outpoint}i0`;
+}
+
+export function decodePSBT({ hex, base64, buffer }: OneOfAllDataFormats): Psbt {
+  if (hex) return Psbt.fromHex(hex);
+  if (base64) return Psbt.fromBase64(base64);
+  if (buffer) return Psbt.fromBuffer(buffer);
+
+  throw new OrditSDKError("Invalid options");
+}
+
+export function decodeTx({ hex, buffer }: BufferOrHex): Transaction {
+  if (hex) return Transaction.fromHex(hex);
+  if (buffer) return Transaction.fromBuffer(buffer);
+
+  throw new OrditSDKError("Invalid options");
+}
+
+function isPaymentFactory(payment: PaymentCreator, network: Network) {
```

Comment:
I wonder if this would work without network. I'm checking bitcoinjs lib, they have the same function, only difference is they passing the network

Reply:
i checked that they don't pass `network` there, maybe bitcoinjs-lib only allows mainnet?

@iamcrazycoder was `network` manually added because on bitcoinjs-lib, it only supported one network?

Reply:
If network is not passed, it defaults to `mainnet`. 

bitcoinjs-lib supports all networks. We have added the network here because the script, address, and all related data in the resulting payload would be different for each network.

Reply:
@thedoublejay Resolving this since this seems to be used internally within ordit-sdk and we added Network